iterators.tex

```diff
@@ -6394,26 +6397,32 @@
 only if \tcode{size(t)} returns the number of elements in \tcode{t}.}
 
 \pnum
-For any type \tcode{T}, \tcode{is_sized_range<T>} derives from
-\tcode{true_type} if \tcode{T} \oldtxt{models}\newtxt{meets the syntactic requirements of}
-\tcode{\xname{SizedRangeLike}}, and \tcode{false_type} otherwise.
+For a\oldtxt{ny} type \tcode{T} \newtxt{that satisfies \tcode{\xname{SizedRangeLike<T>}}},
+\tcode{is_sized_range<T>} derives from \tcode{true_type} if \oldtxt{\tcode{T} models
+\tcode{\xname{SizedRangeLike}}} \newtxt{\tcode{Same<T, remove\_cv\_t<remove\_reference\_t<T>{}>{}>()} is satisfied}, and \oldtxt{\tcode{false_type}} \newtxt{\tcode{is_sized_range<remove_cv_t<remove_reference_t<T>{}>{}>}} otherwise.
```

Comment:
I see what you did there. That isn't too awful, I suppose. The bit with `meta::has_type` could also be written `_Valid<meta::_t, enable_view<__uncvref<T>>>`. For the specification, we can't refer to `meta`, though. *&lt;sigh&gt;* C++ needs some metaprogramming facilities.

> Although that clause with the disjunctions is hideous, and probably needs to become a predicate for performance. 

Why not put the derived-from/!container-like logic into the `enable_view` trait?

Another approach I've sometimes used in cases like this is make the primary template inherit from a special type. That way, there's a simple way to detect whether the template has been specialized. Not sure if that helps here.

FWIW, I think I'm now convinced.